ddclient.in

```diff
@@ -570,6 +570,20 @@ my %services = (
             'server' => setv(T_FQDNP, 1, 0, 'freemyip.com', undef),
         },
     },
+    'gandi' => {
+        'updateable' => undef,
+        'update'     => \&nic_gandi_update,
+        'examples'   => \&nic_gandi_examples,
+        'variables'  => {
+            %{$variables{'service-common-defaults'}},
+            'min-interval' => setv(T_DELAY,  0, 0, 0,                  interval('5m')),
+            'server'       => setv(T_FQDNP,  1, 0, 'api.gandi.net/v5', undef),
+            'ttl'          => setv(T_NUMBER, 1, 0, 10800,              300),
+            'zone'         => setv(T_FQDN,   1, 0, '',                 undef),
```

Comment:
The zone can be any subdomain of a TLD. A hostname can be such as `host.example.co.uk` or `host.suba.subb.example.co.uk` in such cases it can be hazardous to guess the host part from the zone one.

Reply:
OK. I imagine that for most users the zone is the FQDN minus the first part, no? If so, it would make sense to have the zone be optional and derived from the FQDN if unset.

Reply:
Well users can register `example.co.uk` but they may want to update `host.example.co.uk` or `host.sub.example.co.uk` in both cases the zone in `example.co.uk` so only taking the last member will lead to errors.